### PR TITLE
Sdk update

### PR DESCRIPTION
**Pull request by kendricktan**
Added:

1. More helper functions in the `sdk` (smart-contracts/helpers)
2. `ThroughProxy` contract functions for compound (i.e. you can directly borrow/supply/redeem/payback your compund assets without transferring assets to your proxy address)

### CODE REVIEW

File: packages/smart-contracts/test/common.ts [hunk 1]
```diff
@@ -49,3 +49,45 @@ export const tryAndWait = async (f: any) => {
     }
   }
 };
+
+export const getTokenFromUniswapAndApproveProxyTransfer = async (
```
Comment (adrianmcli) [inline packages/smart-contracts/test/common.ts:53]: What button does this correspond to?

File: packages/smart-contracts/helpers/proxyFactory.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    enterMarketsCallbackData,
+    {
+      gasLimit
+    }
+  );
+};
+
+export default {
+  buildAndEnterMarkets,
+  buildAndEnterMarketsOwner
```
Comment (adrianmcli) [inline packages/smart-contracts/helpers/proxyFactory.ts:53]: What's the difference between these two?

Comment (kendricktan) [reply packages/smart-contracts/helpers/proxyFactory.ts:53]: One creates it for msg.sender, one creates it for `owner`

File: packages/smart-contracts/test/common.ts (hunk 1 above)
Comment (kendricktan) [reply packages/smart-contracts/test/common.ts:53]: We dont nees to add this feature. Its only for testing

**Pull request merged**